### PR TITLE
Create DOD (Definition of Done) (Issue/2688)

### LINKED ISSUE #2688: Create DOD (Definition of Done)
Scope
---
For transparency reasons it would be helpful to provide a specific DOD (Definition of Done) within our [Mono Repository](https://github.com/porsche-design-system/porsche-design-system) and expose it on the [Storefront](https://designsystem.porsche.com/).

Notes  
---
The DOD should contain all specific requirements (like acceptance criteria / checklist) which needs to be fulfilled when creating or adjusting a PDS component, e.g.:
- Cross Browser Support
- A11y Compliance
- Up to 200% Text Zoom Support
- High Contrast Mode
- Auto Dark Theme
- Print Support
- Tests written and passing
  - Unit Test
  - E2E Test
  - A11y Test
  - VRT Test
- Code Reviewed
- Changelog updated
- Documentation updated
- Security vulnerability scan passing
- ...

Maybe the DOD can be linked or combined with the [contribution guideline](https://designsystem.porsche.com/v3/help/support#issue-and-contribution-management-at-git-hub).

Acceptance criteria
---
- [x] A DOD (Definition of Done) is exposed on the Storefront under section **Must Know**

Subtasks
---
- [x] Task

### PR DESCRIPTION
**Pull request by denyo**
### Pull Request Checklist

- [x] I have read and accept the [Contributor License Agreement](https://opensource.porsche.com/docs/cla)

#### References

- Preview: https://designsystem.porsche.com/issue/2688/must-know/definition-of-done

#### Scope

Short description of the pull request scope

#### Resolved Issue

Resolves #2688

### CODE REVIEW

File: packages/storefront/src/pages/must-know/definition-of-done.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  - 200% text zoom support
+- Unit tests for/with jsdom-polyfill (if modern browser APIs are used that are not available in jsdom)
+- SSR support for Next.js and Remix (unit tested, and visually tested via puppeteer in Chrome with disabled JavaScript)
+- Smoke tests (if adding new assets to our CDN)
+- Security vulnerability scan passes (CodeQL of GitHub)
+- Cross Browser Support (last 2 stable versions of Chrome, Edge, Firefox, iOS and macOS Safari)
+- Documentation updated
+- Changelog updated
+- Code reviewed (via pull request)
+
```
Comment (tatjana-gartner2) [inline packages/storefront/src/pages/must-know/definition-of-done.md:28]: Should we mention UXPin as well?

**Review by tatjana-gartner2 — COMMENTED** [review]
Looks good to me :)

File: packages/components-js/tests/vrt/puppeteer/fixtures/flyout-scaling.1000.png
Comment (stephanschroeter) [inline packages/components-js/tests/vrt/puppeteer/fixtures/flyout-scaling.1000.png:1]: Is the "Sticky Heading" correct in terms of line-height?

File: packages/storefront/src/pages/must-know/definition-of-done.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+<TableOfContents></TableOfContents>
+
+## Current Criteria
+
+- Unit tests (in node/jsdom)
+- End-to-End Tests (E2E, via puppeteer in Chrome)
+  - relevant functionality
+  - a11y compliance (a11y tree snapshots)
+- Visual Regression Tests (VRT, via puppeteer in Chrome and playwright in Desktop Safari)
```
Comment (stephanschroeter) [inline packages/storefront/src/pages/must-know/definition-of-done.md:15]:
```suggestion
- Visual Regression Tests (VRT, via Puppeteer in Chromium (Chrome) and Playwright in WebKit (Safari))
```

File: packages/storefront/src/pages/must-know/definition-of-done.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  - different prop configurations per component
+  - :hover, :focus and :hover:focus states
+  - high contrast mode
+  - 200% text zoom support
+- Unit tests for/with jsdom-polyfill (if modern browser APIs are used that are not available in jsdom)
+- SSR support for Next.js and Remix (unit tested, and visually tested via puppeteer in Chrome with disabled JavaScript)
```
Comment (stephanschroeter) [inline packages/storefront/src/pages/must-know/definition-of-done.md:21]:
```suggestion
- SSR support for Next.js and Remix (unit tested, and visually tested via Puppeteer  with disabled JavaScript)
```

File: packages/storefront/src/pages/must-know/definition-of-done.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# Definition Of Done
+
+We value high quality and transparency.  
+Thus we consider a task or ticket only done when all of the following criteria are fulfilled as our **Definition of
+Done** (DOD).
+
+<TableOfContents></TableOfContents>
+
+## Current Criteria
+
```
Comment (stephanschroeter) [inline packages/storefront/src/pages/must-know/definition-of-done.md:10]: Maybe we should also add things like, code is written with: 
- Semantic HTML
- Smallest possible DOM hierarchy
- Modern CSS techniques producing as less CSS as possible
- In general having everything properly typed, abstracted, tested and maintainable
…

File: packages/storefront/src/pages/must-know/definition-of-done.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  - Proper typing, abstraction, reusability and maintainability
+- Code is tested with
+  - Unit tests (in node/jsdom)
+  - End-to-End Tests (E2E, via puppeteer in Chrome)
+  - Visual Regression Tests (VRT, via Puppeteer in Chromium and Playwright in WebKit browser engines)
+  - Unit tests for/with jsdom-polyfill (if modern browser APIs are used that are not available in jsdom)
+  - Smoke tests (if adding new assets to our CDN)
+- Accessibility compliance (WCAG 2.1 level AA)
+  - High Contrast Mode
+  - 200% Text Zoom Support
```
Comment (stephanschroeter) [inline packages/storefront/src/pages/must-know/definition-of-done.md:26]:
```suggestion
- Accessibility compliance
  - WCAG 2.1 level AA
  - High Contrast Mode
  - 200% Text Zoom Support
```
To make clear, that high contrast + text-zoom is something additional.

File: packages/storefront/src/pages/must-know/definition-of-done.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  - Smallest possible DOM hierarchy
+  - Modern CSS techniques producing as few CSS as possible
+  - Performance at runtime
+  - Proper typing, abstraction, reusability and maintainability
+- Code is tested with
+  - Unit tests (in node/jsdom)
+  - End-to-End Tests (E2E, via puppeteer in Chrome)
+  - Visual Regression Tests (VRT, via Puppeteer in Chromium and Playwright in WebKit browser engines)
+  - Unit tests for/with jsdom-polyfill (if modern browser APIs are used that are not available in jsdom)
+  - Smoke tests (if adding new assets to our CDN)
```
Comment (stephanschroeter) [inline packages/storefront/src/pages/must-know/definition-of-done.md:23]:
```suggestion
- Code is tested with
  - Unit tests (in node/jsdom)
  - End-to-End Tests (E2E, via puppeteer in Chrome)
  - Accessibility Tests (A11y, via Axe Core)
  - Visual Regression Tests (VRT, via Puppeteer in Chromium and Playwright in WebKit browser engines)
  - Unit tests for/with jsdom-polyfill (if modern browser APIs are used that are not available in jsdom)
  - Smoke tests (if adding new assets to our CDN)
```

**Review by stephanschroeter — APPROVED** [review]

**Pull request merged**